src/float_impl.rs

```diff
@@ -270,3 +271,25 @@ impl<F: Float + FloatConst, C: FloatChecker<F>> FloatConst for NoisyFloat<F, C>
     #[inline] fn PI() -> Self { Self::new(F::PI()) }
     #[inline] fn SQRT_2() -> Self { Self::new(F::SQRT_2()) }
 }
+
+impl<F: Float, C: FloatChecker<F>> iter::Sum for NoisyFloat<F, C> {
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = Self>,
+    {
+        // the sum of a finite list of reals must be real,
+        // and the sum of a finite list of non-NaN must be non-NaN
```

Comment:
This is not true. For example, `f64::MAX + f64::MAX` is `f64::INFINITY`. Also, `f64::INFINITY + f64::NEG_INFINITY` is `NaN`. Furthermore, the `NoisyFloat` trait may be implemented with a user-defined checker, and you can't account for all of the possibilities there. There are similar problems with multiplication, for example, `f64::MAX * f64::MAX * 0.0` is `NaN`.

I'll accept the code review if you replace the `unchecked_new` with `new`, here and below.

Reply:
Oops, you're right; I was thinking from a math perspective, not a programming one. The commit below fixes this.